lib/tasks/import.rake

```
@@ -31,6 +32,97 @@ namespace :import do
     end
     puts "\nDone"
   end
+
+  desc "Import datasets from a data.gov.uk dump"
+  task :datasets, [:filename] => :environment do |_, args|
+
+    # Maps the organisation UUIDs to the organisation IDs
+    orgs_cache =  Organisation.all.pluck(:uuid, :id).to_h
+    counter = 0
+
+    json_from_lines(args.filename) do |obj|
+      counter += 1
+      if counter % 500 == 0
+        puts "Completed #{counter}"
+      end
+
+      d = Dataset.find_or_create_by(name: obj["name"])
+      d.uuid = obj["id"]
+      d.title = obj["title"]
+      d.summary = generate_summary(obj["notes"])
+      d.description = obj["notes"]
+      d.organisation_id = orgs_cache[obj["owner_org"]]
+      d.licence = obj["license_id"]
+      d.licence = 'no-licence' if d.licence == ""
+      d.licence_other = ""
+      d.frequency = convert_frequency(obj["update_frequency"])
+      d.published = true
+      d.published_date = obj["metadata_created"]
+      d.created_at = obj["metadata_created"]
+      d.updated_at = obj["metadata_modified"]
+      d.dataset_type = dataset_type(obj)
+      d.harvested = harvested?(obj)
+      d.location1 = ""
+      d.location2 = ""
+      d.location3 = ""
+      d.legacy_metadata = ""
+      d.save()
```

save! so it fails loudly